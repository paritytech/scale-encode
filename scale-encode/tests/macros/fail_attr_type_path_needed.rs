use scale_encode::encode_as_type;

// A type_path is needed to know what type to
// generate the impl for when attr macro is used.
#[encode_as_type]
struct Foo {
    ty: u8,
    out: bool,
    context: String,
    types: u64
}

fn main() {}