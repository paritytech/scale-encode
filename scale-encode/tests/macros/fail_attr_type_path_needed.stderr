error: The #[encode_as_type] attribute macro requires that #[encode_as_type(type_path = "::path::to::Type")] is given
 --> tests/macros/fail_attr_type_path_needed.rs:5:1
  |
5 | #[encode_as_type]
  | ^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the attribute macro `encode_as_type` (in Nightly builds, run with -Z macro-backtrace for more info)
